charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml

```diff
@@ -3,17 +3,32 @@ groups:
   rules:
   # Recording rules for pod usage
   - record: seed:container_cpu_usage_seconds_total:sum_by_pod
-    expr: sum(rate(container_cpu_usage_seconds_total[5m])) by (pod, namespace)
+    expr: sum(rate(container_cpu_usage_seconds_total[5m])) by (pod, container, namespace)
 
   - record: seed:container_memory_working_set_bytes:sum_by_pod
-    expr: sum(container_memory_working_set_bytes) by (pod, namespace)
+    expr: sum(container_memory_working_set_bytes) by (pod, container, namespace)
 
   - record: seed:container_network_receive_bytes_total:sum_by_pod
-    expr: sum(rate(container_network_receive_bytes_total[5m])) by (pod, namespace)
+    expr: sum(rate(container_network_receive_bytes_total[5m])) by (pod, container, namespace)
 
   - record: seed:container_network_transmit_bytes_total:sum_by_pod
     expr: sum(rate(container_network_transmit_bytes_total[5m])) by (pod, namespace)
 
+  # Recording rules for pod requests
+  - record: seed:kube_pod_container_resource_requests_cpu_cores:sum_by_pod
+    expr: sum(kube_pod_container_resource_requests_cpu_cores) by (pod, container, namespace)
+
+  - record: seed:kube_pod_container_resource_requests_memory_bytes:sum_by_pod
+    expr: sum(kube_pod_container_resource_requests_memory_bytes) by (pod, container, namespace)
+
+  # Recording rules for vpa recommendations
+  - record: seed:vpa_status_recommendation:sum_by_target
+    expr: sum(vpa_status_recommendation{recommendation="target"}) by (targetName, container, resource, namespace)
```

Comment:
This recording rule needs to be part of the aggregate-prometheus recording rules. Then you will only have vpa information of vpa objects in the garden ns. Either the aggregate prometheus needs to federate the data from each prometheus or get all metrics from the vpa exporter.

Reply:
👍 I was testing these and couldn't see these metrics. I will make the changes you suggested and give it another test.